ds-ml/weather/weather-pipeline.py

```
@@ -0,0 +1,106 @@
+    
+    return weather_df
+
+def fetch_irish_holidays(start_year=2023, end_year=None):
+    if end_year is None:
+        end_year = datetime.now().year
+    
+    holidays_list = []
+    
+    for year in range(start_year, end_year + 1):
+        url = f"https://date.nager.at/api/v3/publicholidays/{year}/IE"
+        response = requests.get(url)
```

Ruff (0.13.2)

Probable use of `requests` call without timeout

(S113)